File: redsheet-portal/src/app/database-models/email-template.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { EmailTemplateDetail,  } from './'
export class EmailTemplate {
    constructor(emailTemplate?: EmailTemplate )  {
        let properties = [ "body", "emailTemplateId", "emailTemplateName", "isActive", "statusId", "subject", "emailTemplateDetails",];
        for (let property of properties)
            if (emailTemplate && emailTemplate[property])
                this[property] = emailTemplate[property];
    }
 
    @required()
	body : string =   undefined;
 
	emailTemplateId : number =   0 ;
 
    @required()
    @maxLength(100)
	emailTemplateName : string =   undefined;
 
	isActive : boolean = false ;
 
    @range(1,2147483647)
	statusId : number =   undefined;
 
    @required()
    @maxLength(200)
	subject : string =   undefined;
	@nested(EmailTemplateDetail)
	emailTemplateDetails: EmailTemplateDetail[];



}

File: RedSheet.DbEntities/Models/EmailTemplate.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("EmailTemplates", Schema = "dbo")]
	[Serializable]
	public partial class EmailTemplate  
	{
 
	[Required]
	public string Body { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int EmailTemplateId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string EmailTemplateName { get; set; } 
 
	[Required]
	public bool IsActive { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int StatusId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string Subject { get; set; } 
	[InverseProperty("EmailTemplate")]
	public ICollection<EmailTemplateDetail> EmailTemplateDetails { get; set; } 


	public EmailTemplate()
	{
	this.EmailTemplateDetails = new HashSet<EmailTemplateDetail>();

    }

    }
}

